SDK/MobileCenter/Microsoft.Azure.Mobile.UWP/Utils/DefaultScreenSizeProvider.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Windows.Foundation.Metadata;
+using Windows.Graphics.Display;
+
+namespace Microsoft.Azure.Mobile.Utils
+{
+    class DefaultScreenSizeProvider : AbstractScreenSizeProvider
+    {
+        private readonly SemaphoreSlim _displayInformationEventSemaphore = new SemaphoreSlim(0);
+        private readonly object _lockObject = new object();
+
+        // Either of these == -1 translates to screen size of "unknown"
+        private int _cachedScreenHeight = -1;
+        private int _cachedScreenWidth = -1;
+
+        private const string FailureMessage = "Could not determine display size.";
+
+        public DefaultScreenSizeProvider()
+        {                
+            if (!ApiInformation.IsPropertyPresent(typeof(DisplayInformation).FullName, "ScreenHeightInRawPixels") ||
+                !ApiInformation.IsPropertyPresent(typeof(DisplayInformation).FullName, "ScreenWidthInRawPixels"))
+            {
+                MobileCenterLog.Warn(MobileCenterLog.LogTag, FailureMessage);
+                _displayInformationEventSemaphore.Release();
+                return;
+            }
+            try
+            {
+                // CurrentSynchronization context is essentially the UI context, which is needed
+                // to get display information. It isn't guaranteed to be available, so try/catch
+                var context = TaskScheduler.FromCurrentSynchronizationContext();
+                Task.Factory.StartNew(() =>
```

Comment:
Yes, need to use task.factory.startnew to provide the synchronization context so that the delegate is run on the UI thread, which is not possible with Task.Run